pyminio/tests/tests.py

```diff
@@ -0,0 +1,312 @@
+from os.path import join
+
+import pytest
+
+from .utils import PyminioTest
+
+ROOT = '/'
+FILE_CONTENT = bytes('test', 'UTF-8')
+
+
+def _mkdirs_recursively(client, tree, relative_path):
+    for key, value in tree.items():
+        abspath = join(relative_path, key)
+        if value is None:
+            client.mkdirs(join(relative_path, ''))
+            client.put_data(abspath, FILE_CONTENT)
+
+        elif value == []:
+            client.mkdirs(join(abspath, ''))
+
+        else:
+            _mkdirs_recursively(
+                client=client,
+                tree=tree[key],
+                relative_path=abspath)
+
+
+def mkdirs_recursively(tree, relative_path=ROOT):
+    def decorator(func):
+        def wrapper(client):
+            _mkdirs_recursively(
+                client=client,
+                tree=tree,
+                relative_path=relative_path)
+            return func(client)
+        return wrapper
+    return decorator
+
+
+@pytest.fixture
+def client():
+    client = PyminioTest()
+    yield client
+    client.rm('/', recursive=True)
+
+
+@mkdirs_recursively(
+    tree={
+        'foo': {
+            'bar': {
+                'baz': None
+            }
+        }
+    }
```

Comment:
i think a better name should be:
```
@mock_fs({
'foo': {
  'bar': {
     'baz': None
  }
})
```

no need for tree=
and also fs or structure is better than tree I think.